### PR TITLE
Add linter

### PR DESCRIPTION
**Pull request by discombobulateme**
In this PR we added: ESLint, Prettier and Husky to add pre-commit error messages. 
That intends to create a more conforming Docs.

### CODE REVIEW

**Review by aoleszyn — APPROVED** [review]

**Review by lysannep — APPROVED** [review]

**Comment by lysannep** [comment]
@spider-sauce @discombobulateme ~ I currently use a spell checker and the Prettier extension for Visual Studio Code, will that cause any conflicts or issues?

**Comment by spider-sauce** [comment]
@lysannep we added the `--fix` flag for `eslint` so aside from any manual fixes that would break the build most formatting issues are enforced through the config. It shouldn't affect your extensions because it will only enforce those standards after you commit. If you're interested in which rules we enforce, you can look at the `.eslintrc` , and also the `.prettierrc` file. Let us know if you do encounter any overwritten rules or issues and we can adjust from there.

**Review by wswebcreation — CHANGES_REQUESTED** [review]
Hi @discombobulateme 

I'm really happy with this! But, there is a but.. 😉 . I would have expected that also we would see changes for the markdown files (my biggest concern every time I save my files in Visual Studio). If I would now open `/docs/mobile-apps/automated-testing/appium/configuration.md` and hit save I see this

image

So my linter in VSCode is linting the markdown files, but when I run `npx eslint . --fix` nothing happens. This is related to the fact that ESLint can only lint JS files and no Markdown files. If I would for example run the following command ` npx eslint . --fix && npx prettier --check './docs/**/*.md'` I get a lot of warnings because now prettier is being used to check the markdown files.
By adding the changing `--check` into `--write` it will also alter the files

<details>
  <summary>Click to see/hide logs</summary>
```logs
npx eslint . --fix && npx prettier --check './docs/**/*.md'                 
npm WARN config init.license Use `--init-license` instead.
npm WARN config init.license Use `--init-license` instead.
Checking formatting...
[warn] docs/api-testing.md
[warn] docs/api-testing/build-from-spec.md
[warn] docs/api-testing/composer.md
[warn] docs/api-testing/composer/assertion-components.md
[warn] docs/api-testing/composer/expressions.md
[warn] docs/api-testing/composer/io-components.md
[warn] docs/api-testing/composer/logical-components.md
[warn] docs/api-testing/composer/other-components.md
[warn] docs/api-testing/contract-testing.md
[warn] docs/api-testing/environments.md
[warn] docs/api-testing/import-export-tests.md
[warn] docs/api-testing/import-postman-collection.md
[warn] docs/api-testing/integrations/apifctl-cicd-integration.md
[warn] docs/api-testing/integrations/pagerduty-webhooks.md
[warn] docs/api-testing/legacy.md
[warn] docs/api-testing/load-testing.md
[warn] docs/api-testing/logger.md
[warn] docs/api-testing/mocking.md
[warn] docs/api-testing/on-prem/api/using-the-api.md
[warn] docs/api-testing/on-prem/api/v3.md
[warn] docs/api-testing/on-prem/assertion-components/assert-compares.md
[warn] docs/api-testing/on-prem/assertion-components/assert-contains.md
[warn] docs/api-testing/on-prem/assertion-components/assert-equals.md
[warn] docs/api-testing/on-prem/assertion-components/assert-exists.md
[warn] docs/api-testing/on-prem/assertion-components/assert-greater.md
[warn] docs/api-testing/on-prem/assertion-components/assert-in.md
[warn] docs/api-testing/on-prem/assertion-components/assert-is.md
[warn] docs/api-testing/on-prem/assertion-components/assert-less.md
[warn] docs/api-testing/on-prem/assertion-components/assert-matches.md
[warn] docs/api-testing/on-prem/assertion-components/assert-valid-json-schema.md
[warn] docs/api-testing/on-prem/bloodhound.md
[warn] docs/api-testing/on-prem/bloodhound/advanced-actors.md
[warn] docs/api-testing/on-prem/bloodhound/base-actors.md
[warn] docs/api-testing/on-prem/bloodhound/basic-configuration.md
[warn] docs/api-testing/on-prem/bloodhound/expressions.md
[warn] docs/api-testing/on-prem/bloodhound/fine-tuning.md
[warn] docs/api-testing/on-prem/bloodhound/flows.md
[warn] docs/api-testing/on-prem/bloodhound/forward-calls-to-apif-logger.md
[warn] docs/api-testing/on-prem/bloodhound/load-balancing.md
[warn] docs/api-testing/on-prem/bloodhound/module-actors.md
[warn] docs/api-testing/on-prem/bloodhound/module-actors/fortress-forwarder.md
[warn] docs/api-testing/on-prem/bloodhound/module-actors/jdbc.md
[warn] docs/api-testing/on-prem/bloodhound/module-actors/mongodb.md
[warn] docs/api-testing/on-prem/bloodhound/module-actors/rabbitmq.md
[warn] docs/api-testing/on-prem/bloodhound/using-databases.md
[warn] docs/api-testing/on-prem/ci/apif-auto.md
[warn] docs/api-testing/on-prem/ci/azure-devops.md
[warn] docs/api-testing/on-prem/ci/bitbucket.md
[warn] docs/api-testing/on-prem/ci/connecting-with-bamboo.md
[warn] docs/api-testing/on-prem/ci/gitlab.md
[warn] docs/api-testing/on-prem/ci/jenkins/apif-auto-and-github.md
[warn] docs/api-testing/on-prem/ci/jenkins/apif-auto.md
[warn] docs/api-testing/on-prem/ci/jenkins/using-the-api.md
[warn] docs/api-testing/on-prem/ci/jenkins/zephyr-enterprise-integration.md
[warn] docs/api-testing/on-prem/ci/micro-focus-alm-integration.md
[warn] docs/api-testing/on-prem/how-to/3-leg-oauth.md
[warn] docs/api-testing/on-prem/how-to/apf-local-engine.md
[warn] docs/api-testing/on-prem/how-to/assertions-for-metrics-performance.md
[warn] docs/api-testing/on-prem/how-to/authorization-simple-oauth-etc.md
[warn] docs/api-testing/on-prem/how-to/copy-a-test.md
[warn] docs/api-testing/on-prem/how-to/create-a-dynamic-header.md
[warn] docs/api-testing/on-prem/how-to/different-ways-to-compose-a-request-body.md
[warn] docs/api-testing/on-prem/how-to/disable-ssl-validation.md
[warn] docs/api-testing/on-prem/how-to/executing-from-cucumber.md
[warn] docs/api-testing/on-prem/how-to/following-redirects.md
[warn] docs/api-testing/on-prem/how-to/footprint.md
[warn] docs/api-testing/on-prem/how-to/generate-a-status-page.md
[warn] docs/api-testing/on-prem/how-to/generating-fake-test-data.md
[warn] docs/api-testing/on-prem/how-to/github-for-datasets.md
[warn] docs/api-testing/on-prem/how-to/import-export-tests.md
[warn] docs/api-testing/on-prem/how-to/integration-testing.md
[warn] docs/api-testing/on-prem/how-to/keystores-for-downloader.md
[warn] docs/api-testing/on-prem/how-to/kv.md
[warn] docs/api-testing/on-prem/how-to/load-agent-deployment.md
[warn] docs/api-testing/on-prem/how-to/set-dynamic-dates.md
[warn] docs/api-testing/on-prem/how-to/setup-notifications-performance-alerts.md
[warn] docs/api-testing/on-prem/how-to/testing-graphql.md
[warn] docs/api-testing/on-prem/how-to/tweaking-stateful-email-notifications.md
[warn] docs/api-testing/on-prem/how-to/update-input.md
[warn] docs/api-testing/on-prem/how-to/using-long-lasting-tokens.md
[warn] docs/api-testing/on-prem/how-to/version-control.md
[warn] docs/api-testing/on-prem/how-to/writing-tests-in-code.md
[warn] docs/api-testing/on-prem/integrations/add-new-connector.md
[warn] docs/api-testing/on-prem/integrations/alerts-thresholds-using-json-connector.md
[warn] docs/api-testing/on-prem/integrations/build-from-apiary.md
[warn] docs/api-testing/on-prem/integrations/build-from-axway.md
[warn] docs/api-testing/on-prem/integrations/build-from-mashery.md
[warn] docs/api-testing/on-prem/integrations/connectors/bigpanda-io.md
[warn] docs/api-testing/on-prem/integrations/connectors/elastic-kibana.md
[warn] docs/api-testing/on-prem/integrations/connectors/jira.md
[warn] docs/api-testing/on-prem/integrations/connectors/pagerduty.md
[warn] docs/api-testing/on-prem/integrations/connectors/setup-connectors-datadog.md
[warn] docs/api-testing/on-prem/integrations/connectors/splunk.md
[warn] docs/api-testing/on-prem/integrations/connectors/statuspage.md
[warn] docs/api-testing/on-prem/integrations/connectors/twilio.md
[warn] docs/api-testing/on-prem/integrations/connectors/xmatters.md
[warn] docs/api-testing/on-prem/integrations/helper-databases-jdbc.md
[warn] docs/api-testing/on-prem/integrations/jenkins.md
[warn] docs/api-testing/on-prem/integrations/single-sign-on-with-saml-2-0-beta.md
[warn] docs/api-testing/on-prem/io-components.md
[warn] docs/api-testing/on-prem/io-components/jdbc.md
[warn] docs/api-testing/on-prem/io-components/multipart-upload.md
[warn] docs/api-testing/on-prem/learn-more/downloader-101.md
[warn] docs/api-testing/on-prem/learn-more/improving-your-metrics.md
[warn] docs/api-testing/on-prem/learn-more/input-set-and-global-variables.md
[warn] docs/api-testing/on-prem/learn-more/planning-your-test.md
[warn] docs/api-testing/on-prem/learn-more/the-variables-system-in-api-fortress.md
[warn] docs/api-testing/on-prem/learn-more/the-working-copy-published-tests.md
[warn] docs/api-testing/on-prem/learn-more/working-with-the-response-object.md
[warn] docs/api-testing/on-prem/logical-components/each.md
[warn] docs/api-testing/on-prem/logical-components/flow.md
[warn] docs/api-testing/on-prem/logical-components/if.md
[warn] docs/api-testing/on-prem/logical-components/tag.md
[warn] docs/api-testing/on-prem/logical-components/while.md
[warn] docs/api-testing/on-prem/mocking/expressions-in-mocked-apis.md
[warn] docs/api-testing/on-prem/mocking/mock-recording-with-kong.md
[warn] docs/api-testing/on-prem/postman/build-from-collections.md
[warn] docs/api-testing/on-prem/product-information/api-tests-vs-schema-validation.md
[warn] docs/api-testing/on-prem/product-information/cloud-vs-on-premises-local.md
[warn] docs/api-testing/on-prem/quick-start.md
[warn] docs/api-testing/on-prem/quick-start/adding-a-user.md
[warn] docs/api-testing/on-prem/quick-start/build-from-spec.md
[warn] docs/api-testing/on-prem/quick-start/composer.md
[warn] docs/api-testing/on-prem/quick-start/dashboard.md
[warn] docs/api-testing/on-prem/quick-start/easy-monitoring.md
[warn] docs/api-testing/on-prem/quick-start/environments-vault-and-overrides-magic.md
[warn] docs/api-testing/on-prem/quick-start/flexible-variables-for-flexible-environments.md
[warn] docs/api-testing/on-prem/quick-start/forge.md
[warn] docs/api-testing/on-prem/quick-start/importing-postman-collections.md
[warn] docs/api-testing/on-prem/quick-start/integrate-with-your-cicd.md
[warn] docs/api-testing/on-prem/quick-start/introduction-to-integration-testing.md
[warn] docs/api-testing/on-prem/quick-start/load-testing.md
[warn] docs/api-testing/on-prem/quick-start/mocking-services.md
[warn] docs/api-testing/on-prem/quick-start/schedule-a-test.md
[warn] docs/api-testing/on-prem/quick-start/setup-connectors.md
[warn] docs/api-testing/on-prem/quick-start/test-reports.md
[warn] docs/api-testing/on-prem/quick-start/the-vault.md
[warn] docs/api-testing/on-prem/quick-start/using-the-example-snippets.md
[warn] docs/api-testing/on-prem/reference/comment.md
[warn] docs/api-testing/on-prem/reference/company-settings.md
[warn] docs/api-testing/on-prem/reference/composer-snippets.md
[warn] docs/api-testing/on-prem/reference/connecting-to-testrail.md
[warn] docs/api-testing/on-prem/reference/execution-context-in-api-fortress.md
[warn] docs/api-testing/on-prem/reference/expression-language-extensions.md
[warn] docs/api-testing/on-prem/reference/expression.md
[warn] docs/api-testing/on-prem/reference/generating-a-jwt.md
[warn] docs/api-testing/on-prem/reference/github.md
[warn] docs/api-testing/on-prem/reference/read-file.md
[warn] docs/api-testing/on-prem/security-testing/example-security-tests.md
[warn] docs/api-testing/on-prem/self-hosted/best-practices-for-disaster-recovery.md
[warn] docs/api-testing/on-prem/self-hosted/certificate-based-mutual-ssl-tls-authentication.md
[warn] docs/api-testing/on-prem/self-hosted/deployment-configure-the-dns-for-the-mocking-service.md
[warn] docs/api-testing/on-prem/self-hosted/deployment-docker.md
[warn] docs/api-testing/on-prem/self-hosted/deployment-kubernetes.md
[warn] docs/api-testing/on-prem/self-hosted/enabling-api-fortress-to-read-local-files.md
[warn] docs/api-testing/on-prem/self-hosted/local-downloader-setup.md
[warn] docs/api-testing/on-prem/self-hosted/multiple-client-cert-downloader.md
[warn] docs/api-testing/on-prem/self-hosted/on-prem-platform.md
[warn] docs/api-testing/on-prem/self-hosted/on-premises-backing-up-your-data.md
[warn] docs/api-testing/on-prem/self-hosted/on-premises-requirements.md
[warn] docs/api-testing/on-prem/self-hosted/proxy-settings-in-downloader.md
[warn] docs/api-testing/on-prem/self-hosted/red-hat-openshift.md
[warn] docs/api-testing/on-prem/self-hosted/updating-an-on-premises-instance.md
[warn] docs/api-testing/on-prem/self-hosted/updating-the-api-fortress-license-key.md
[warn] docs/api-testing/on-prem/self-hosted/using-rds-and-documentdb.md
[warn] docs/api-testing/project-access.md
[warn] docs/api-testing/project-dashboard.md
[warn] docs/api-testing/quickstart.md
[warn] docs/api-testing/sauce-connect.md
[warn] docs/api-testing/schedule-test.md
[warn] docs/api-testing/variables-environment-overrides.md
[warn] docs/api-testing/vault.md
[warn] docs/basics/acct-team-mgmt-hub.md
[warn] docs/basics/acct-team-mgmt/adding-deactivating-users.md
[warn] docs/basics/acct-team-mgmt/adding-deleting-teams.md
[warn] docs/basics/acct-team-mgmt/assigning-removing-users-teams.md
[warn] docs/basics/acct-team-mgmt/concurrency-limits.md
[warn] docs/basics/acct-team-mgmt/managing-subscription.md
[warn] docs/basics/acct-team-mgmt/managing-user-info.md
[warn] docs/basics/acct-team-mgmt/org-settings.md
[warn] docs/basics/acct-team-mgmt/plan-details.md
[warn] docs/basics/acct-team-mgmt/private-device-mgmt.md
[warn] docs/basics/acct-team-mgmt/sauce-connect-proxy-tunnels.md
[warn] docs/basics/acct-team-mgmt/test-object.md
[warn] docs/basics/acct-team-mgmt/updating-billing.md
[warn] docs/basics/acct-team-mgmt/viewing-exporting-usage-data.md
[warn] docs/basics/data-center-endpoints.md
[warn] docs/basics/data-center-endpoints/aust-early-access.md
[warn] docs/basics/environment-variables.md
[warn] docs/basics/integrations-overview.md
[warn] docs/basics/integrations/deque.md
[warn] docs/basics/integrations/eggplant.md
[warn] docs/basics/integrations/evinced.md
[warn] docs/basics/integrations/jira.md
[warn] docs/basics/integrations/slack.md
[warn] docs/basics/integrations/sumo.md
[warn] docs/basics/integrations/tosca.md
[warn] docs/basics/integrations/uipath.md
[warn] docs/basics/integrations/webhooks.md
[warn] docs/basics/platform-configurator.md
[warn] docs/basics/quickstarts.md
[warn] docs/basics/sso-hub.md
[warn] docs/basics/sso/config-adfs.md
[warn] docs/basics/sso/config-okta.md
[warn] docs/basics/sso/setting-up-single-sign-on.md
[warn] docs/basics/test-config-annotation/test-annotation.md
[warn] docs/basics/test-config-annotation/test-config.md
[warn] docs/ci/azure.md
[warn] docs/ci/bamboo.md
[warn] docs/ci/gitlab.md
[warn] docs/ci/jenkins.md
[warn] docs/ci/teamcity.md
[warn] docs/contributing.md
[warn] docs/contributing/adv-examples.md
[warn] docs/contributing/code-of-conduct.md
[warn] docs/contributing/style-guide.md
[warn] docs/dev/api.md
[warn] docs/dev/api/accounts.md
[warn] docs/dev/api/api-testing.md
[warn] docs/dev/api/builds.md
[warn] docs/dev/api/connect.md
[warn] docs/dev/api/insights.md
[warn] docs/dev/api/jobs.md
[warn] docs/dev/api/performance.md
[warn] docs/dev/api/platform.md
[warn] docs/dev/api/rdc.md
[warn] docs/dev/api/storage.md
[warn] docs/dev/cli.md
[warn] docs/dev/cli/sauce-connect-proxy.md
[warn] docs/dev/cli/saucectl.md
[warn] docs/dev/cli/saucectl/architecture.md
[warn] docs/dev/cli/saucectl/completion.md
[warn] docs/dev/cli/saucectl/configure.md
[warn] docs/dev/cli/saucectl/init.md
[warn] docs/dev/cli/saucectl/run.md
[warn] docs/dev/cli/saucectl/storage/download.md
[warn] docs/dev/cli/saucectl/storage/list.md
[warn] docs/dev/cli/saucectl/storage/upload.md
[warn] docs/dev/cli/saucectl/usage/ci/circleci.md
[warn] docs/dev/cli/saucectl/usage/ci/github-actions.md
[warn] docs/dev/cli/saucectl/usage/ci/gitlab.md
[warn] docs/dev/cli/saucectl/usage/ci/jenkins.md
[warn] docs/dev/cli/saucectl/usage/ide/intellij.md
[warn] docs/dev/cli/saucectl/usage/ide/vscode.md
[warn] docs/dev/cli/saucectl/usage/use-cases.md
[warn] docs/dev/cli/virtual-usb.md
[warn] docs/dev/cli/virtual-usb/connect-session.md
[warn] docs/dev/cli/virtual-usb/delete-session.md
[warn] docs/dev/cli/virtual-usb/disconnect-session.md
[warn] docs/dev/cli/virtual-usb/find-sessionid.md
[warn] docs/dev/cli/virtual-usb/start-server.md
[warn] docs/dev/cli/virtual-usb/start-session.md
[warn] docs/dev/data-center-maint.md
[warn] docs/dev/error-messages.md
[warn] docs/dev/glossary.md
[warn] docs/dev/low-code.md
[warn] docs/dev/low-code/analyze-step.md
[warn] docs/dev/low-code/dashboard.md
[warn] docs/dev/low-code/execute/reports-page.md
[warn] docs/dev/low-code/execute/schedules-page.md
[warn] docs/dev/low-code/nlp-reference.md
[warn] docs/dev/low-code/plan/projects/project-details-page.md
[warn] docs/dev/low-code/plan/projects/projects.md
[warn] docs/dev/low-code/plan/test-cases.md
[warn] docs/dev/low-code/plan/test-suites.md
[warn] docs/dev/low-code/profile.md
[warn] docs/dev/low-code/troubleshooting.md
[warn] docs/dev/low-code/variables.md
[warn] docs/dev/test-configuration-options.md
[warn] docs/error-reporting/getting-started.md
[warn] docs/error-reporting/platform-integrations/android/configuration.md
[warn] docs/error-reporting/platform-integrations/android/proguard-deobfuscation.md
[warn] docs/error-reporting/platform-integrations/android/setup.md
[warn] docs/error-reporting/platform-integrations/ios/configuration.md
[warn] docs/error-reporting/platform-integrations/ios/setup.md
[warn] docs/error-reporting/platform-integrations/real-devices/setup.md
[warn] docs/error-reporting/platform-integrations/unity/attributes.md
[warn] docs/error-reporting/platform-integrations/unity/configuration.md
[warn] docs/error-reporting/platform-integrations/unity/setup.md
[warn] docs/error-reporting/platform-integrations/unreal/attributes.md
[warn] docs/error-reporting/platform-integrations/unreal/setup.md
[warn] docs/error-reporting/platform-integrations/unreal/troubleshooting.md
[warn] docs/error-reporting/project-setup/access-control.md
[warn] docs/error-reporting/project-setup/alerts.md
[warn] docs/error-reporting/project-setup/attributes.md
[warn] docs/error-reporting/project-setup/data-retention.md
[warn] docs/error-reporting/project-setup/deduplication-rules.md
[warn] docs/error-reporting/project-setup/deduplication.md
[warn] docs/error-reporting/project-setup/file-attachments.md
[warn] docs/error-reporting/project-setup/metric-export.md
[warn] docs/error-reporting/project-setup/metrics-stability-scores.md
[warn] docs/error-reporting/project-setup/object-reprocessing.md
[warn] docs/error-reporting/project-setup/scheduled-reports.md
[warn] docs/error-reporting/project-setup/source-code.md
[warn] docs/error-reporting/project-setup/submission-actions.md
[warn] docs/error-reporting/project-setup/symbol-servers.md
[warn] docs/error-reporting/project-setup/symbolication.md
[warn] docs/error-reporting/project-setup/team-mgmnt.md
[warn] docs/error-reporting/project-setup/user-mgmnt.md
[warn] docs/error-reporting/security-compliance/common-questions.md
[warn] docs/error-reporting/security-compliance/privacy-control.md
[warn] docs/error-reporting/web-console/debug.md
[warn] docs/error-reporting/web-console/explore.md
[warn] docs/error-reporting/web-console/flame-graphs.md
[warn] docs/error-reporting/web-console/getting-started.md
[warn] docs/error-reporting/web-console/overview.md
[warn] docs/error-reporting/web-console/releases.md
[warn] docs/error-reporting/web-console/triage.md
[warn] docs/error-reporting/whats-new.md
[warn] docs/error-reporting/workflow-integrations/alerting-ops/opsgenie.md
[warn] docs/error-reporting/workflow-integrations/alerting-ops/pagerduty.md
[warn] docs/error-reporting/workflow-integrations/alerting-ops/victorops.md
[warn] docs/error-reporting/workflow-integrations/common-settings.md
[warn] docs/error-reporting/workflow-integrations/issue-tracking/asana.md
[warn] docs/error-reporting/workflow-integrations/issue-tracking/fogbugz.md
[warn] docs/error-reporting/workflow-integrations/issue-tracking/github.md
[warn] docs/error-reporting/workflow-integrations/issue-tracking/jira.md
[warn] docs/error-reporting/workflow-integrations/issue-tracking/phabricator.md
[warn] docs/error-reporting/workflow-integrations/messaging/amazon-sns.md
[warn] docs/error-reporting/workflow-integrations/messaging/email.md
[warn] docs/error-reporting/workflow-integrations/messaging/gitter.md
[warn] docs/error-reporting/workflow-integrations/messaging/irc.md
[warn] docs/error-reporting/workflow-integrations/messaging/slack.md
[warn] docs/error-reporting/workflow-integrations/messaging/stride.md
[warn] docs/error-reporting/workflow-integrations/monitoring/circonus.md
[warn] docs/error-reporting/workflow-integrations/monitoring/datadog.md
[warn] docs/error-reporting/workflow-integrations/overview.md
[warn] docs/error-reporting/workflow-integrations/webhook.md
[warn] docs/headless.md
[warn] docs/insights/coverage.md
[warn] docs/insights/debug.md
[warn] docs/insights/failure-analysis.md
[warn] docs/insights/history.md
[warn] docs/insights/scope.md
[warn] docs/insights/trends.md
[warn] docs/mobile-apps.md
[warn] docs/mobile-apps/app-storage.md
[warn] docs/mobile-apps/automated-testing.md
[warn] docs/mobile-apps/automated-testing/appium.md
[warn] docs/mobile-apps/automated-testing/appium/appium-2-migration.md
[warn] docs/mobile-apps/automated-testing/appium/appium-flutter.md
[warn] docs/mobile-apps/automated-testing/appium/configuration.md
[warn] docs/mobile-apps/automated-testing/appium/quickstart.md
[warn] docs/mobile-apps/automated-testing/appium/real-devices.md
[warn] docs/mobile-apps/automated-testing/appium/virtual-devices.md
[warn] docs/mobile-apps/automated-testing/espresso-xcuitest.md
[warn] docs/mobile-apps/automated-testing/espresso-xcuitest/espresso-capture.md
[warn] docs/mobile-apps/automated-testing/espresso-xcuitest/espresso.md
[warn] docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest.md
[warn] docs/mobile-apps/automated-testing/ipa-files.md
[warn] docs/mobile-apps/faq.md
[warn] docs/mobile-apps/features/audio-capture.md
[warn] docs/mobile-apps/features/biometric-authentication.md
[warn] docs/mobile-apps/features/bypass-screenshot.md
[warn] docs/mobile-apps/features/camera-image-injection.md
[warn] docs/mobile-apps/features/gestures.md
[warn] docs/mobile-apps/features/mobile-app-diagnostics/device-vitals.md
[warn] docs/mobile-apps/features/mobile-app-diagnostics/interactions.md
[warn] docs/mobile-apps/features/mobile-app-diagnostics/view-tree.md
[warn] docs/mobile-apps/features/network-capture.md
[warn] docs/mobile-apps/features/virtual-usb.md
[warn] docs/mobile-apps/live-testing/live-mobile-app-testing.md
[warn] docs/mobile-apps/ms-app-center.md
[warn] docs/mobile-apps/real-device-cleaning.md
[warn] docs/mobile-apps/supported-devices.md
[warn] docs/overview.md
[warn] docs/performance.md
[warn] docs/performance/about.md
[warn] docs/performance/analyze.md
[warn] docs/performance/motion.md
[warn] docs/performance/one-page.md
[warn] docs/performance/transitions.md
[warn] docs/sauce-basics.md
[warn] docs/secure-connections.md
[warn] docs/secure-connections/ipsec-vpn.md
[warn] docs/secure-connections/sauce-connect.md
[warn] docs/secure-connections/sauce-connect/advanced/architecture.md
[warn] docs/secure-connections/sauce-connect/advanced/kgp.md
[warn] docs/secure-connections/sauce-connect/advanced/specifications.md
[warn] docs/secure-connections/sauce-connect/changelog.md
[warn] docs/secure-connections/sauce-connect/faq.md
[warn] docs/secure-connections/sauce-connect/installation.md
[warn] docs/secure-connections/sauce-connect/proxy-tunnels.md
[warn] docs/secure-connections/sauce-connect/quickstart.md
[warn] docs/secure-connections/sauce-connect/security-authentication.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/additional-proxies.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/basic-setup.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/ci-cd-integration.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/docker.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/environment-variables.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/high-availability.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/specialized-environments.md
[warn] docs/secure-connections/sauce-connect/setup-configuration/yaml-config.md
[warn] docs/secure-connections/sauce-connect/system-requirements.md
[warn] docs/secure-connections/sauce-connect/troubleshooting.md
[warn] docs/test-results.md
[warn] docs/test-results/archived-test-results.md
[warn] docs/test-results/badges-browser-matrix.md
[warn] docs/test-results/sharing-test-results.md
[warn] docs/test-results/test-status.md
[warn] docs/test-results/viewing-test-results.md
[warn] docs/visual.md
[warn] docs/visual/acct-team-mgmt.md
[warn] docs/visual/component-testing/integrations/continuous-integration.md
[warn] docs/visual/component-testing/integrations/github.md
[warn] docs/visual/component-testing/integrations/sauce-labs.md
[warn] docs/visual/component-testing/integrations/slack.md
[warn] docs/visual/component-testing/integrations/visual-studio-team-services.md
[warn] docs/visual/component-testing/integrations/webhooks.md
[warn] docs/visual/component-testing/setup.md
[warn] docs/visual/component-testing/storybook-interactions-testing.md
[warn] docs/visual/component-testing/storybook-static.md
[warn] docs/visual/component-testing/supported-browsers.md
[warn] docs/visual/component-testing/workflow/baseline-branch.md
[warn] docs/visual/component-testing/workflow/change-details.md
[warn] docs/visual/component-testing/workflow/ignoring-changes.md
[warn] docs/visual/component-testing/workflow/include-exclude-settings.md
[warn] docs/visual/component-testing/workflow/review-workflow.md
[warn] docs/visual/component-testing/workflow/visual-history.md
[warn] docs/visual/css-animations.md
[warn] docs/visual/e2e-testing/code-examples.md
[warn] docs/visual/e2e-testing/commands-options.md
[warn] docs/visual/e2e-testing/integrations/continuous-integration.md
[warn] docs/visual/e2e-testing/integrations/slack.md
[warn] docs/visual/e2e-testing/integrations/webhooks.md
[warn] docs/visual/e2e-testing/setup.md
[warn] docs/visual/e2e-testing/supported-browsers.md
[warn] docs/visual/e2e-testing/troubleshooting.md
[warn] docs/visual/e2e-testing/workflow/baseline-branch.md
[warn] docs/visual/e2e-testing/workflow/change-details.md
[warn] docs/visual/e2e-testing/workflow/ignoring-changes.md
[warn] docs/visual/e2e-testing/workflow/review-workflow.md
[warn] docs/visual/e2e-testing/workflow/visual-history.md
[warn] docs/visual/faq.md
[warn] docs/visual/notifications.md
[warn] docs/web-apps.md
[warn] docs/web-apps/automated-testing.md
[warn] docs/web-apps/automated-testing/_partials/_advanced-cypress.md
[warn] docs/web-apps/automated-testing/_partials/_advanced-playwright.md
[warn] docs/web-apps/automated-testing/_partials/_advanced.md
[warn] docs/web-apps/automated-testing/cucumberjs-playwright/advanced.md
[warn] docs/web-apps/automated-testing/cucumberjs-playwright/quickstart.md
[warn] docs/web-apps/automated-testing/cucumberjs-playwright/yaml.md
[warn] docs/web-apps/automated-testing/cypress.md
[warn] docs/web-apps/automated-testing/cypress/advanced.md
[warn] docs/web-apps/automated-testing/cypress/quickstart.md
[warn] docs/web-apps/automated-testing/cypress/yaml.md
[warn] docs/web-apps/automated-testing/cypress/yaml/v1.md
[warn] docs/web-apps/automated-testing/cypress/yaml/v1alpha.md
[warn] docs/web-apps/automated-testing/playwright.md
[warn] docs/web-apps/automated-testing/playwright/advanced.md
[warn] docs/web-apps/automated-testing/playwright/quickstart.md
[warn] docs/web-apps/automated-testing/playwright/yaml.md
[warn] docs/web-apps/automated-testing/puppeteer.md
[warn] docs/web-apps/automated-testing/puppeteer/quickstart.md
[warn] docs/web-apps/automated-testing/puppeteer/yaml.md
[warn] docs/web-apps/automated-testing/replay.md
[warn] docs/web-apps/automated-testing/replay/quickstart.md
[warn] docs/web-apps/automated-testing/replay/yaml.md
[warn] docs/web-apps/automated-testing/selenium.md
[warn] docs/web-apps/automated-testing/selenium/pre-run-executables.md
[warn] docs/web-apps/automated-testing/selenium/quickstart.md
[warn] docs/web-apps/automated-testing/selenium/sample-scripts.md
[warn] docs/web-apps/automated-testing/selenium/selenium-grid.md
[warn] docs/web-apps/automated-testing/selenium/selenium4.md
[warn] docs/web-apps/automated-testing/selenium/troubleshooting.md
[warn] docs/web-apps/automated-testing/testcafe.md
[warn] docs/web-apps/automated-testing/testcafe/advanced.md
[warn] docs/web-apps/automated-testing/testcafe/quickstart.md
[warn] docs/web-apps/automated-testing/testcafe/yaml.md
[warn] docs/web-apps/live-testing/live-cross-browser-testing.md
[warn] Code style issues found in 471 files. Forgot to run Prettier?
```
</details>

For markdown we can also ignore stuff, see [this](https://prettier.io/blog/2017/11/07/1.8.0.html).

I'm not 100% sure how everybody feels about also linting the markdown code, but I'm a huge fan of it so it would not result in a lot of space/line breaks and so on and enforce some styling.
If we would go for it then the change might be big, so we really need to test it. We also need to determine if we want to run ESLint and Prettier, as mentioned above, as two separate commands, or only go for Prettier.
Then we also need to add some examples for people if they have different editors
Last but not least, my suggestion would be to not automatically fix the changes on commit, but make it break the commit and let people manually validate the changes. There might be situations where the linting BREAKS the implementation and manual changes are needed to for example ignore the linter on some places

**Comment by spider-sauce** [comment]
@wswebcreation 

TL;DR I think all of these suggestions are good and valid, it may take some time to cross train before we implement them.

I think scanning markdown files is a good approach from a content predictability standpoint, I just worry about bottlenecking the tech content team (which I'm no longer a member/core maintainer). I'll have to set up some time to cross train, create examples based on which IDE they're all using, common troubleshooting tips, etc. before we implement this. 

I also want them to weigh in on which rules to enforce using prettier because they are creating the majority of the commits and I want to make sure we conform to their style guides too. 

As for prettier vs. eslint, can't we use both? I doubt the tech content team will touch the JS files that much, but if they need any frontend work it would be good to enforce these rules either way. Also, I agree we should remove the `--fix` flag, and we can include that troubleshooting process in the cross training.

**Comment by spider-sauce** [comment]
Hey all,

I did some quick testing with `prettier` and here are the results:

* basic markdown formatting: https://github.com/saucelabs/sauce-docs/pull/1584/files#diff-5190347e972083d2fd4bfda093482c8787f7baedbe41c95d5d93525a3848ed96
* markdown formatting with JSX tabs: https://github.com/saucelabs/sauce-docs/pull/1584/files#diff-29c1825f272bcaddb0e62b31857b5f8237b2b194a2bec75c766337fc8f4ea0b5

I've yet to add the command to `.husky`, @wswebcreation will `npx prettier . --check` prevent the commit if it throws `warn`? I just want to make sure if we opt out of using `--write` flag that it will force folks to manually fix the formatting errors.

**Comment by wswebcreation** [comment]
Hi @spider-sauce 

> As for prettier vs. eslint, can't we use both? I doubt the tech content team will touch the JS files that much, but if they need any frontend work it would be good to enforce these rules either way. 

Yes we can, prettier is more for formatting, eslint is more for code quality checks, because we have more markdown files I thought only prettier was enough, but after doing some research is better to keep them both

> ...will npx prettier . --check prevent the commit if it throws warn?
Based on the [docs](https://prettier.io/docs/en/cli.html#--check) it should return exit code 1, which would stop the push

**Review by spider-sauce — APPROVED** [review]
